File: src/ReetCode/Easy/1to100/Q83/Q83_Remove Duplicates from Sorted List.py
"""
# Author    : Lee In Bok 
# Date      : 2023.11.08(Wed)
# Spend Time: 18m 40s
# Runtime   : 46 ms (Beats 58.11%)
# Memory    : 16.3 MB (Beats 53.09%)
# Algoritm  : Linked List
"""

# Definition for singly-linked list.
# class ListNode:
#     def __init__(self, val=0, next=None):
#         self.val = val
#         self.next = next

# class Solution:
#     def deleteDuplicates(self, head: Optional[ListNode]) -> Optional[ListNode]:
#         prev = None
#         temp = head

#         while temp:
#             if prev and prev.val == temp.val:
#                 prev.next = temp.next
#             else:
#                 prev = temp
            
#             temp = temp.next

#         return head

File: CodeTest/src/ReetCode/Easy/1to100/Q83/Q83_RemoveDuplicatesfromSortedList.java
/**
* Author    : Lee In Bok 
* Date      : 2023.11.08(Wed)
* Spend Time: 18m 40s
* Runtime   : 0 ms (Beats 100%)
* Memory    : 42.6 MB (Beats 87.50%)
* Algoritm  : Linked List
 */

 /**
 * Definition for singly-linked list.
 * public class ListNode {
 *     int val;
 *     ListNode next;
 *     ListNode() {}
 *     ListNode(int val) { this.val = val; }
 *     ListNode(int val, ListNode next) { this.val = val; this.next = next; }
 * }
 */

// class Solution {
//     public ListNode deleteDuplicates(ListNode head) {
//         ListNode temp = head;
//         ListNode prev = null;

//         while(temp != null){
//             if(prev != null && prev.val == temp.val){
//                 prev.next = temp.next;
//             } else {
//                 prev = temp;
//             }

//             temp = temp.next;
//         }

//         return head;
//     }
// }